sunpy/net/dataretriever/sources/goes.py

```diff
@@ -1,18 +1,58 @@
-#Author: Rishabh Sharma <rishabh.sharma.gunner@gmail.com>
-#This module was developed under funding provided by
-#Google Summer of Code 2014
+"""
+This module implements GOES Client.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
 
 import datetime
+import urllib2
+from bs4 import BeautifulSoup
 
-from sunpy.time import parse_time, TimeRange
+from sunpy.time import TimeRange
+from sunpy.util.scraper import Scraper
 
 from ..client import GenericClient
 
 __all__ = ['GOESClient']
 
 
 class GOESClient(GenericClient):
+    """
+    Returns a list of URLS to SOLIS VSM files corresponding to value of input timerange.
+    URL source: `http://gong2.nso.edu/pubkeep/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument - 'goes' ,case insensitive.
 
+    Physobs: Optional, includes 'INTENSITY', "IRRADIANCE'
+             'PARTICLE_FLUX'.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> res = Fido.search(a.Time('2016/6/4 00:00:00','2016/6/4 00:03:00'),
+                Instrument('goes'), a.Physobs('INTENSITY'))
+    >>> print (res)
+    [<Table length=7>
+         Start Time           End Time      Source Instrument
+           str19               str19         str4     str4
+    ------------------- ------------------- ------ ----------
+    2016-06-04 00:00:00 2016-06-05 00:00:00   nasa       goes
+    2016-06-05 00:00:00 2016-06-06 00:00:00   nasa       goes
+    2016-06-06 00:00:00 2016-06-07 00:00:00   nasa       goes
+    2016-06-07 00:00:00 2016-06-08 00:00:00   nasa       goes
+    2016-06-08 00:00:00 2016-06-09 00:00:00   nasa       goes
+    2016-06-09 00:00:00 2016-06-10 00:00:00   nasa       goes
+    2016-06-10 00:00:00 2016-06-11 00:00:00   nasa       goes]
```

Comment:
You mentioned once that the tables aren't outputting the right results. I looked at the downloading code in ```dataretriever```. [This](https://github.com/sudk1896/sunpy/blob/unidown/sunpy/net/dataretriever/client.py#L80) is responsible for generating the table output. I would have to look at the core downloading code to make sense of the table output. It will take some time.